src/__tests__/unit/repository/soft-crud.repository.unit.ts

```diff
@@ -288,6 +288,38 @@ describe('SoftCrudRepository', () => {
         expect(e.message).to.be.equal('EntityNotFound');
       }
     });
+    it('should not return soft deleted entry by id, without using deleted in fields filter', async () => {
+      try {
+        await repo.findById(3, {
+          fields: {
+            id: true,
+            email: true,
+          },
+        });
+        fail();
+      } catch (e) {
+        expect(e.message).to.be.equal('EntityNotFound');
+      }
+    });
+    it('should return requested fields only when not using deleted in fields filter', async () => {
+      const customer = await repo.findById(4, {
+        fields: {
+          id: true,
+          email: true,
+        },
+      });
+      expect(customer).to.not.have.property('deleted');
+    });
+    it('should return requested fields matched with fields filter', async () => {
+      const customer = await repo.findById(4, {
+        fields: {
+          id: true,
+          email: true,
+          deleted: true,
+        },
+      });
+      expect(customer).to.have.property('deleted');
```

Comment:
still should not return deleted row, fields filter means "properties to include or exclude from the results." it doesn't mean to return deleted row.. deleted entry should be returned only if requested by 'where deleted=true'

Reply:
deleted entries should never be returned. The test case showcases the scenario if the deleted property is requested in the entity object or not. If it is requested then the property should be present otherwise not. And the id I used here is not soft deleted. @vaibhavkumar-sf

Reply:
ok, got it

Reply:
This is correct. @prernagp can you check ?

Reply:
Sir, I am not returning deleted rows. This test case is written to return deleted property or not as per user request.

Reply:
ok